File: asgmt_001/psuedoinverse.jl
using LinearAlgebra

x_arr = [1 0; 1 1; 1 2; 1 3;]
y_arr = [0; 1; 4; 9;]

global total = 0
for i  in 1:10
  global total
  start_time = time_ns()
  x_inv = pinv(x_arr)
  end_results = x_inv * y_arr
  total += (time_ns() - start_time)
  println(end_results)
end

println("Average Time in Julia: ",total/10000000000)

File: asgmt_001/psuedoinverse.py
import numpy as np
import time

x_arr = np.array([[1,0], [1,1], [1,2], [1,3]])
y_arr = np.array([[0], [1], [4], [9]])

total = 0
for i  in range(10):
  start_time = time.time()
  x_inv = np.linalg.pinv(x_arr)
  end_results = np.dot(x_inv, y_arr)
  total += (time.time() - start_time)
  print(end_results)

print("Average Time in Python: ",total/10)
